### PR TITLE
up eslint packages + type check and format in pre-commit

### PR DESCRIPTION
**Pull request by technophile-04**
### Description:

Updated pre-commit hook so that it does:

- Type check: By doing this we make sure that type error don't get into our code.
  - eslint / typescript-eslint doesn't do typescheck [ref](https://github.com/typescript-eslint/typescript-eslint/issues/1037) and that's why [this happened](https://github.com/scaffold-eth/create-eth/pull/155#pullrequestreview-2450343240)
- lint check: we were already doing this.
- format: This is helpful for json files, by default eslint does not format json files, but prettier is able to do it nicely. This way we make sure people don't commit unformatted json files especially `extensions.json` file

Update gh-action `lint.yml` file to do typecheck too

---

Also updated the eslint related packages because on the main branch in `eslint.config.js` I was getting this:

image

And also on `main` branch a small error at line in `eslint.config.js` where we were using `eslintPluginPrettierRecommended` (not able to reproduce now though)

### CODE REVIEW

File: lefthook.yml [hunk 1]
```diff
@@ -1,5 +1,14 @@
 pre-commit:
+  parallel: true
   commands:
+    type-check:
+      glob: "src/**/*.{ts,tsx}"
```
Comment (rin-st) [inline lefthook.yml:5]: nit: all tsx files are excluded in tsconfig, right?

**Comment by rin-st** [comment]
Not sure what I'm doing wrong but pre-commit doesn't work for me at all, I can add whatever I want to any files, like `asdfasdf`, even in `main` branch. Do you know what can cause it? Files are not ignored, for example any ts file inside `src` folder

--- 

Also, I'm getting an error when linting

```
yarn lint

/Users/liana/Documents/prog/buidl-guidl/create-eth/src/utils/system-validation.ts
  7:12  error  'error' is defined but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

File: lefthook.yml (hunk 1 above)
Comment (technophile-04) [reply lefthook.yml]: Ohh yes we actually don't need to check for tsx since we won't be having them in cli 🙌 nice catch, thanks Rinat!

**Comment by technophile-04** [comment]
> Not sure what I'm doing wrong but lefthook and hence pre-commit doesn't work for me at all, I can add whatever I want to any files, like asdfasdf, even in main branch. Do you know what can cause it? Files are not ignored, for example any .ts file inside src folder

Umm not sure what's the problem is, but can re-clone the repo and then  `yarn install` and then switching this branch and see if it works? 

Also maybe try running : 

```shell
yarn lefthook run pre-commit
```

To see if lefhook is working?

**Comment by rin-st** [comment]
Lol today it works. I didn't change anything, just restarted mac yesterday but not sure it could be cause of the fix :suspect: 

Now everything works as expected! Lgtm, thank you!

**Review by rin-st — APPROVED** [review]

**Pull request merged**